libraries/AP_HAL_ChibiOS/hwdef/common/bouncebuffer.c

```diff
@@ -100,6 +99,9 @@ void bouncebuffer_finish_read(struct bouncebuffer_t *bouncebuffer, const uint8_t
 {
     if (bouncebuffer && buf == bouncebuffer->dma_buf) {
         osalDbgAssert((bouncebuffer->busy == true), "bouncebuffer finish_read");
+#if defined(STM32H7)
+        stm32_cacheBufferInvalidate(buf, (size+31)&~31);
```

Comment:
surely this can only matter if some other thread reads from these cached lines in between?

Reply:
@tridge - according to the docs I am, in fact, wrong - https://www.chibios.org/dokuwiki/doku.php?id=chibios:articles:cache_coherency
My worry is that I found that this was not true in bdshot despite what the docs say. Given that only complete lack of caching fixes my issue I will close this